x-pack/plugins/fleet/public/applications/fleet/sections/epm/screens/detail/use_get_enriched_package_policies.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useMemo, useState } from 'react';
+import { AgentPolicy, PackagePolicy } from '../../../../../../../common/types/models';
+import {
+  GetAgentPoliciesResponse,
+  GetAgentPoliciesResponseItem,
+} from '../../../../../../../common/types/rest_spec';
+import { useGetPackagePolicies } from '../../../../hooks/use_request';
+import {
+  SendConditionalRequestConfig,
+  useConditionalRequest,
+} from '../../../../hooks/use_request/use_request';
+import { agentPolicyRouteService } from '../../../../../../../common/services';
+import { AGENT_POLICY_SAVED_OBJECT_TYPE } from '../../../../../../../common/constants';
+import { GetPackagePoliciesResponse } from '../../../../../../../common/types/rest_spec';
+
+export interface PackagePolicyEnriched extends PackagePolicy {
+  _agentPolicy: GetAgentPoliciesResponseItem | undefined;
+}
+
+type GetEnrichedPackagePoliciesResponse = Exclude<GetPackagePoliciesResponse, 'items'> & {
+  items: PackagePolicyEnriched[];
+};
+
+/**
+ * Works similar to `useGetAgentPolicies()`, except that it will add an additional property to
+ * each package policy named `_agentPolicy` which may hold the Agent Policy associated with the
+ * given package policy.
+ * @param query
+ */
+export const useGetEnrichedPackagePolicies = (
```

Comment:
Maybe it's more a naming issue here, but I found that hook kind of confusing, and I am wondering if we should:

use two hooks `useGetPackagePolicy` and `useAgentPolicy` when needed

or 

maybe rename this hook usePackagePoliciesWithAgentPolicy  that return `(..., items: [{ agentPolicy, packagePolicy }]`

Reply:
Thats a good suggestion @nchaulet . So instead of adding a property to `PackagePolicy` change the array items to be an object that has two properties: one holds the PackagePolicy and the other holds the associated AgentPolicy.
Will work on getting those changes in. Thanks.